### PR TITLE
asset check result

### PR DESCRIPTION
**Pull request by johannkm**
First pass at the NT that users will yield/return

### CODE REVIEW

**Comment by johannkm** [comment]
Current dependencies on/for this PR:
* master
  * **PR #15839** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/15839" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/15839?utm_source=stack-comment).

File: python_modules/dagster/dagster/_core/definitions/asset_checks.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+class AssetCheckResult(
+    NamedTuple(
+        "_AssetCheckResult",
+        [
+            ('asset_key', Optional[AssetKey]),
+            ('success', bool),
+            ('metadata', List[Mapping[str, MetadataValue]]),
+        ]
+    )):
+    def __new__(cls, asset_key, success, metadata):
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/asset_checks.py:20]: type annotations plz

File: python_modules/dagster/dagster/_core/definitions/asset_checks.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+
+@experimental
+@whitelist_for_serdes
+class AssetCheckResult(
+    NamedTuple(
+        "_AssetCheckResult",
+        [
+            ('asset_key', Optional[AssetKey]),
+            ('success', bool),
+            ('metadata', List[Mapping[str, MetadataValue]]),
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/asset_checks.py:17]: This should be "Mapping[str, MetadataValue]`

File: python_modules/dagster/dagster/_core/definitions/asset_checks.py [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+from dagster._annotations import experimental
+from dagster._core.definitions.events import AssetKey, MetadataValue
+from dagster._serdes import whitelist_for_serdes
+
+
+@experimental
+@whitelist_for_serdes
+class AssetCheckResult(
+    NamedTuple(
+        "_AssetCheckResult",
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/asset_checks.py:13]: These params should probably all be `PublicAttrs`

File: python_modules/dagster/dagster/_core/definitions/asset_checks.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import dagster._check as check
+from dagster._annotations import experimental
+from dagster._core.definitions.events import AssetKey, MetadataValue
+from dagster._serdes import whitelist_for_serdes
+
+
+@experimental
+@whitelist_for_serdes
+class AssetCheckResult(
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/definitions/asset_checks.py:11]: docstring plz

File: python_modules/dagster/dagster/_core/definitions/asset_checks.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+from typing import List, Mapping, NamedTuple, Optional
```
[the remark this answers is not in the document]
Comment (sryza) [reply python_modules/dagster/dagster/_core/definitions/asset_checks.py:1]: makes it easy to remember where to import it from

File: python_modules/dagster/dagster/_core/definitions/asset_check_result.py [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+            ('metadata', PublicAttr[Mapping[str, MetadataValue]]),
+        ]
+    )):
+    def __new__(cls, asset_key:AssetKey, check_name: str, success: bool, metadata: Optional[Mapping[str, MetadataValue]] = None):
+        return super().__new__(
+            cls,
+            asset_key=check.inst_param(asset_key, 'asset_key', AssetKey),
+            check_name=check.str_param(check_name, 'check_name'),
+            success=check.bool_param(success, 'success'),
+            metadata=check.opt_mapping_param(metadata, 'metadata', key_type=str, value_type=MetadataValue),
```
Comment (johannkm) [inline python_modules/dagster/dagster/_core/definitions/asset_check_result.py:27]: made metadata optional

**Review by sryza — APPROVED** [review]
Looks good!

**Pull request merged**